File: app/src/main/java/space/pal/sig/old/repository/service/HubbleService.java
package space.pal.sig.old.repository.service;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import space.pal.sig.old.repository.dto.FeedDto;
import space.pal.sig.old.repository.dto.NewsDto;
import space.pal.sig.old.repository.dto.NewsPreviewDto;

/**
 * SpaceNow
 * Created by Catalin on 7/19/2020
 **/
public interface HubbleService {

    @GET("/api/v3/news")
    Call<List<NewsPreviewDto>> getHubbleFeed(@Query("page") int page);

    @GET("/api/v3/news?page=all")
    Call<List<NewsPreviewDto>> getHubbleFeed();

    @GET("/api/v3/news_release/{id}")
    Call<NewsDto> getHubbleNews(@Path("id") String newsId);

    @GET("/api/v3/external_feed/esa_feed")
    Call<List<FeedDto>> getEuropeanSpaceAgencyFeed(@Query("page") int page);

    @GET("/api/v3/external_feed/esa_feed?page=all")
    Call<List<FeedDto>> getEuropeanSpaceAgencyFeed();

    @GET("/api/v3/external_feed/jwst_feed")
    Call<List<FeedDto>> getJamesWebbSpaceTelescopeFeed(@Query("page") int page);

    @GET("/api/v3/external_feed/jwst_feed?page=all")
    Call<List<FeedDto>> getJamesWebbSpaceTelescopeFeed();

    @GET("/api/v3/external_feed/st_live")
    Call<List<FeedDto>> getSpaceTelescopeLiveFeed(@Query("page") int page);

    @GET("/api/v3/external_feed/st_live?page=all")
    Call<List<FeedDto>> getSpaceTelescopeLiveFeed();

}

File: app/src/main/java/space/pal/sig/network/HubbleService.kt
package space.pal.sig.network

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import space.pal.sig.model.dto.NewsDto
import space.pal.sig.model.dto.NewsPreviewDto

/**
 * SpaceNow
 * Created by Catalin on 12/25/2020
 **/
interface HubbleService {

    @GET("news")
    fun getHubbleFeed(@Query("page") page: Int): Single<List<NewsPreviewDto>>

    @GET("news?page=all")
    fun getHubbleFeed(): Single<List<NewsPreviewDto>>

    @GET("news_release/{id}")
    fun getHubbleNews(@Path("id") newsId: String): Single<NewsDto>

    @GET("external_feed/esa_feed")
    fun getEuropeanSpaceAgencyFeed(@Query("page") page: Int): Single<List<NewsDto>>

    @GET("external_feed/esa_feed?page=all")
    fun getEuropeanSpaceAgencyFeed(): Single<List<NewsDto>>

    @GET("external_feed/jwst_feed")
    fun getJamesWebbSpaceTelescopeFeed(@Query("page") page: Int): Single<List<NewsDto>>

    @GET("external_feed/jwst_feed?page=all")
    fun getJamesWebbSpaceTelescopeFeed(): Single<List<NewsDto>>

    @GET("external_feed/st_live")
    fun getSpaceTelescopeLiveFeed(@Query("page") page: Int): Single<List<NewsDto>>

    @GET("external_feed/st_live?page=all")
    fun getSpaceTelescopeLiveFeed(): Single<List<NewsDto>>

}